deploy/local/dockerfiles/server.Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM php:8.3.6
+
+# install laravel dependencies including composer & npm
+RUN apt-get update -y &&\
+    apt-get install -y openssl zip unzip git nodejs npm &&\
+    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+
+RUN docker-php-ext-install pdo mysqli pdo_mysql && docker-php-ext-enable pdo_mysql
+
+# set work directory
+WORKDIR /jepret
+
+# set the value of `APP_ENV` to `docker` so it will load
+# `.env.docker` for settings
+ENV APP_ENV=docker
+
+# copy app dependencies and install them
+COPY . .
+RUN composer install && npm install
```

Comment:
Just try it, @farulivan. Don't hesitate to try by yourself & experiment. 👍🏻

However, when I tried your idea, the Laravel threw an error because during `composer install`, it still required the whole source files.

Reply:
Yes, you're right, we need to add `composer install --no-script` to solve that error and I think I need to find out more about this `--no-script` and whether this has any potential side effects on our project. 😅

Reply:
@farulivan, after some digging, I don't think `composer install --no-script` is a good idea to execute because it will skip the scripts written in [here](https://github.com/farulivan/jepret/blob/d9500e8dd8e3e91c55f863258ea78c645ec05ba4/composer.json#L39-L53). 

They are essential for setting up the dependency required by Laravel, especially in a clean-slate environment. So, it's better if we still use `composer install`.

You can also read the details about this issue [here](https://chat.openai.com/share/0bdb2f15-72e8-4be1-8f74-23490c307d9d).